File: backend/src/main/java/com/example/backend/service/AzureStorageService.java
package com.example.backend.service;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.UUID;

@Service
public class AzureStorageService {

    @Value("${app.upload.dir}")
    private String uploadDir;

    public String uploadFile(MultipartFile file) throws IOException {
        // Crear directorio de uploads si no existe
        File uploadDirectory = new File(uploadDir);
        if (!uploadDirectory.exists()) {
            uploadDirectory.mkdirs();
        }

        // Generar nombre único para el archivo
        String originalFilename = file.getOriginalFilename();
        String extension = originalFilename.substring(originalFilename.lastIndexOf("."));
        String newFilename = UUID.randomUUID().toString() + extension;

        // Guardar el archivo
        Path filePath = Paths.get(uploadDir, newFilename);
        Files.copy(file.getInputStream(), filePath);

        return newFilename;
    }

    public byte[] downloadFile(String filename) throws IOException {
        Path filePath = Paths.get(uploadDir, filename);
        return Files.readAllBytes(filePath);
    }

    public void deleteFile(String filename) throws IOException {
        Path filePath = Paths.get(uploadDir, filename);
        Files.deleteIfExists(filePath);
    }

    public boolean fileExists(String filename) {
        Path filePath = Paths.get(uploadDir, filename);
        return Files.exists(filePath);
    }
} 

File: backend/src/main/kotlin/com/example/backend/service/AzureStorageService.kt
package com.example.backend.service

import com.azure.storage.blob.BlobClientBuilder
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.multipart.MultipartFile
import java.io.ByteArrayOutputStream
import java.util.UUID

@Service
class AzureStorageService(
    @Value("\${azure.storage.connection-string}")
    private val connectionString: String,
    
    @Value("\${azure.storage.container-name}")
    private val containerName: String
) {
    fun uploadImage(file: MultipartFile): String {
        val fileName = "${UUID.randomUUID()}.${file.originalFilename?.substringAfterLast('.', "jpg")}"
        
        val blobClient = BlobClientBuilder()
            .connectionString(connectionString)
            .containerName(containerName)
            .blobName(fileName)
            .buildClient()

        blobClient.upload(file.inputStream, file.size)
        
        return fileName
    }

    fun getImage(fileName: String): ByteArray {
        val blobClient = BlobClientBuilder()
            .connectionString(connectionString)
            .containerName(containerName)
            .blobName(fileName)
            .buildClient()

        val outputStream = ByteArrayOutputStream()
        blobClient.downloadStream(outputStream)
        
        return outputStream.toByteArray()
    }

    fun deleteImage(fileName: String) {
        val blobClient = BlobClientBuilder()
            .connectionString(connectionString)
            .containerName(containerName)
            .blobName(fileName)
            .buildClient()

        blobClient.deleteIfExists()
    }
} 